### DISCUSSION TITLE
Include financial functions as part of core or as an added extension

### DISCUSSION THREAD

**Discussion by jdgriffith** [duckdb/duckdb#11769] [Ideas]
Coming from a financial analytics background (retirement portfolios and price action of symbols), I feel duckdb is well positioned to do more in the finance space. I think additional functions could be added around technical analysis of pricing streams (time series data), performance/returns, and more quant-y nuanced functions. I realize some of these functions would either need a proper abstraction or have an opinionated take on the schema required - this could be a hard hurdle to jump in many cases.  

I'm curious what others think of this idea and where it properly fits. Should it be an extension or is there merit for adding more industry specific functions to duckdb core functions?

**Comment by pdet** [comment]
Hi @jdgriffith, in general, the core does not take specialized functions/operators, these do sit better on extensions.

Funny enough, I did build a simple extension with financial functions, although the goal, at the time, was more to have it as an example of how to build extensions.

https://github.com/pdet/Scrooge-McDuck

**Reply by jdgriffith** [reply]
Nice! 

I agree - extensions feel like the appropriate place for this kind of logic.

**Reply by pdet** [reply]
@jdgriffith 
Feel free to open issues with desired functionality in there. I want to revive that extension, the last thing I started to work on was some simple efficient frontier functionality 😅, but since I only touched it on my free time, things may move somewhat slowly (:

**Reply by jdgriffith** [reply]
Thanks! I'm taking a closer look and may use your repo as a baseline :) I will keep you updated in case you want to collab on it some more as time permits.
